topdown/trusted_identity_providers.go

```diff
@@ -0,0 +1,121 @@
+package topdown
+
+import (
+	"errors"
+	"fmt"
+	"github.com/coreos/go-oidc"
+	jwt "github.com/square/go-jose/v3/jwt"
+	"golang.org/x/net/context"
+	"sync"
+	"time"
+)
+
+
+var globalTrustedIdProviderManager TrustedIdProviderManagerImpl
+var globalTrustedIssuerContextTimeout = time.Second * time.Duration(5)
+
+// May be used to manage an token verification against an entire collection
+// of trusted IdP's.
+type TrustedIdProviderManager interface {
```

Comment:
We may want to consider moving this code into its own internal package over in `/internal/` like `/internal/oidc` or something. I think there is enough of this that is generic (ie, not builtin or topdown specific) that we can split it out into a reusable internal package.

Reply:
Very good suggestion, many thx